packages/apollo-collaboration-server/src/fileHandling/fileHandling.service.ts

```diff
@@ -503,4 +508,107 @@ export class FileHandlingService {
     this.logger.debug('GFF3 is not loaded in cache!')
     return false
   }
+
+  /**
+   * This method check that each line has unique id. If not it creates one for each line and overwrites the orignal file
+   * @param filenameWithPath - filename with path
+   */
+  async checkGFF3uniqueKey(filenameWithPath: string) {
+    const stringOfGFF3 = await fs.readFile(filenameWithPath, {
+      encoding: 'utf8',
+      flag: 'r',
+    })
+    this.logger.verbose(`Data read from file=${stringOfGFF3}`)
+
+    const arrayOfThings = gff.parseStringSync(stringOfGFF3, {
+      parseAll: true,
+    })
+    this.logger.debug(`Setting apollo_ids if required....`)
+
+    // Loop all lines and check if each line has 'apollo_id' property inside attributes
+    for (const entry of arrayOfThings) {
+      // Comment, Directive and FASTA -entries are not presented as an array
+      if (Array.isArray(entry)) {
+        for (const [key, val] of Object.entries(entry)) {
+          this.logger.verbose(
+            `GFF3Item: key=${JSON.stringify(key)}, value=${JSON.stringify(
+              val,
+            )}`,
+          )
+          if (val.hasOwnProperty('attributes')) {
+            const assignedVal = Object.assign(val)
```

Comment:
A type should be specified here to TypeScript knows what the type is.

```suggestion
            const assignedVal: GFF3FeatureLineWithRefs = Object.assign(val)
```